lib/shared/sdk/usbboot/index.js

```diff
@@ -409,13 +443,37 @@ exports.scan = (options) => {
       debug(`Serial number index: ${serialNumberIndex}`)
 
       if (serialNumberIndex === USB_DESCRIPTOR_NULL_INDEX) {
+        // eslint-disable-next-line no-magic-numbers
+        emitProgress(10)
         return queryBlobFromCache(USBBOOT_BOOTCODE_FILE_NAME, options.readFile).then((bootcode) => {
           return writeBootCode(device, endpoint, bootcode)
         })
       }
 
       debug('Starting file server')
-      return startFileServer(device, endpoint, options.readFile)
+
+      // eslint-disable-next-line no-magic-numbers
+      emitProgress(15)
+
+      const PERCENTAGE_START = 20
+      const PERCENTAGE_TOTAL = 95
+
+      // TODO: Find a way to not hardcode these values, and instead
+      // figure out the correct number for each board on the fly.
+      // This might be possible once we implement proper device
+      // auto-discovery.
+      // eslint-disable-next-line no-magic-numbers
+      const STEPS_TOTAL = device.deviceDescriptor.idProduct === USB_PRODUCT_ID_BCM2708_BOOT ? 29 : 38
```

Comment:
By this stage, the idProduct will always be `USB_PRODUCT_ID_BCM2710_BOOT` ?
You mentioned in Flowdock that in your tests the USB address stays unique, so I think we need some kind of global lookup table which maps the device-addresses seen in stage1 to their stage1-productID, so that when we get to here during stage2, we can then use the device-address to lookup whether this is actually a CM1 or a CM3?

Also, it might make sense to store these hard-coded numbers (i.e. 29 and 38) in some kind of data-file in the `lib/blobs/usbboot/` directory? Because the number-of-steps depends on both which board we're dealing with, and also the contents of the 'blobs' themselves (specifically config.txt).